src/main/java/com/google/devtools/build/lib/rules/cpp/LinkCommandLine.java

```diff
@@ -311,67 +291,17 @@ public static void extractArgumentsForStaticLinkParamFile(
+      List<String> args, List<String> commandlineArgs, List<String> paramFileArgs) {
     // Note, that it is not important that all linker arguments are extracted so that
     // they can be moved into a parameter file, but the vast majority should.
     commandlineArgs.add(args.get(0)); // gcc command, must not be moved!
     int argsSize = args.size();
-    if (doNotSplitLinkingCmdline) {
-      for (int i = 1; i < argsSize; i++) {
-        String arg = args.get(i);
-        if (arg.startsWith("@")) {
-          commandlineArgs.add(arg); // params file, keep it in the command line
-        } else {
-          paramFileArgs.add(arg); // the rest goes to the params file
-        }
-      }
-    } else {
-      for (int i = 1; i < argsSize; i++) {
-        String arg = args.get(i);
-        if (arg.isEmpty()) {
-          continue;
-        }
-        if (arg.equals("-Wl,-no-whole-archive")) {
-          paramFileArgs.add("-no-whole-archive");
-        } else if (arg.equals("-Wl,-whole-archive")) {
-          paramFileArgs.add("-whole-archive");
-        } else if (arg.equals("-Wl,--start-group")) {
-          paramFileArgs.add("--start-group");
-        } else if (arg.equals("-Wl,--end-group")) {
-          paramFileArgs.add("--end-group");
-        } else if (arg.equals("-Wl,--start-lib")) {
-          paramFileArgs.add("--start-lib");
-        } else if (arg.equals("-Wl,--end-lib")) {
-          paramFileArgs.add("--end-lib");
-        } else if (arg.charAt(0) == '-') {
-          if (arg.startsWith("-l")) {
-            paramFileArgs.add(arg);
-          } else {
-            // Anything else starting with a '-' can stay on the commandline.
-            commandlineArgs.add(arg);
-            if (arg.equals("-o")) {
-              // Special case for '-o': add the following argument as well - it is the output file!
-              commandlineArgs.add(args.get(++i));
-            }
-          }
-        } else if (CppFileTypes.OBJECT_FILE.apply(arg)
-            || CppFileTypes.PIC_OBJECT_FILE.apply(arg)
-            || CppFileTypes.ARCHIVE.apply(arg)
-            || CppFileTypes.PIC_ARCHIVE.apply(arg)
-            || CppFileTypes.ALWAYS_LINK_LIBRARY.apply(arg)
-            || CppFileTypes.ALWAYS_LINK_PIC_LIBRARY.apply(arg)
-            || CppFileTypes.SHARED_LIBRARY.apply(arg)
-            || CppFileTypes.INTERFACE_SHARED_LIBRARY.apply(arg)
-            || CppFileTypes.VERSIONED_SHARED_LIBRARY.apply(arg)) {
-          // All objects of any kind go into the linker parameters.
-          paramFileArgs.add(arg);
-        } else {
-          // Everything that's left stays conservatively on the commandline.
-          commandlineArgs.add(arg);
-        }
+    for (int i = 1; i < argsSize; i++) {
+      String arg = args.get(i);
+      if (arg.startsWith("@")) {
+        commandlineArgs.add(arg); // params file, keep it in the command line
+      } else {
+        paramFileArgs.add(arg); // the rest goes to the params file
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 297:306, 281:290.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.